ethexe/contracts/src/Router.sol

```diff
@@ -383,15 +375,15 @@ contract Router is IRouter, OwnableUpgradeable, ReentrancyGuardTransient {
         Storage storage router = _router();
         require(router.genesisBlock.hash != bytes32(0), "router genesis is zero; call `lookupGenesisHash()` first");
 
-        require(
+        /*require(
             router.protocolData.codes[_codeId] == Gear.CodeState.Validated,
             "code must be validated before program creation"
-        );
+        );*/
+        // TODO: ^^^ uncomment this (for testing only)
 
         // Check for duplicate isn't necessary, because `Clones.cloneDeterministic`
         // reverts execution in case of address is already taken.
-        address actorId =
-            Clones.cloneDeterministic(router.implAddresses.mirrorProxy, keccak256(abi.encodePacked(_codeId, _salt)));
+        address actorId = Clones.cloneDeterministic(address(this), keccak256(abi.encodePacked(_codeId, _salt)));
```

Comment:
you clone self?

Reply:
and why is mirror proxy removed at all?

Reply:
This does not mean cloning itself. The first argument is no longer implementation address, but the Router address. The entire MirrorProxy code can be found in `out/MirrorProxy.sol/MirrorProxy.json` in `deployedBytecode` field. It is completely copied and pasted into our custom `Clones.sol`. The clone program now calls `IRouter(router).mirrorImpl()` in `MirrorProxy` and makes delegate call to the received address.